controllers/topics/getLatestPost.js

```diff
@@ -64,12 +66,32 @@ module.exports = async (req, res) => {
 
     let message = `There are new posts from ${authorUsername}`;
 
+    const getstreamClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+    let unreadCount = 0;
+    try {
+      await getstreamClient.connectUser({id: userId}, token);
+
+      const userChannel = getstreamClient.channel('topics', `topic_${name}`);
+      const createdChannel = await userChannel.watch();
+      const readUsers = createdChannel?.read;
+      const selfUserReadStatus = readUsers?.find((readStatus) => {
+        return readStatus?.user?.id === userId;
+      });
+
+      unreadCount = selfUserReadStatus?.unread_messages || 0;
+    } catch (e) {
+      console.error('Error getting unread count', e);
+    } finally {
+      getstreamClient.disconnectUser();
+    }
+
     res.status(200).json({
       status: 'success',
       code: 200,
       data: {
         ...post,
-        message
+        message,
+        unread_count: unreadCount
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The response object now includes the `unread_count` field. This is a good addition as it provides more information to the client about the state of the topic for the current user. However, ensure that the client-side code is updated to handle this new field in the response object.


<!-- This is an auto-generated comment by OSS CodeRabbit -->